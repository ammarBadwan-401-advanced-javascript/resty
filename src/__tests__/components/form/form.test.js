import React from 'react';
import { shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import Main from '../../../components/form/form';

describe('Form test',()=>{
  it('Buttons exist in the page',()=>{
    let app = shallow(<Main/>);
    expect(app.find('button').exists()).toBeTruthy();
  });

  it('Method changes on button press',()=>{
    let app = mount(<Main/>);
    let button = app.find('button').at(1);
    button.simulate('click');
    console.log(button)
    expect(app.state('method')).toBe('GET');
  });
});
