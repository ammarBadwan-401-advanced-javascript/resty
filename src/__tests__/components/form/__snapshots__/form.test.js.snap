// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form test renders correctly 1`] = `
<main>
  <form>
    <label>
      URL
    </label>
    <input
      onChange={[Function]}
    />
    <button
      onClick={[Function]}
    >
      GO!
    </button>
    <div>
      <button
        onClick={[Function]}
        value="GET"
      >
        GET
      </button>
      <button
        onClick={[Function]}
        value="POST"
      >
        POST
      </button>
      <button
        onClick={[Function]}
        value="PUT"
      >
        PUT
      </button>
      <button
        onClick={[Function]}
        value="DELETE"
      >
        DELETE
      </button>
    </div>
  </form>
  <section
    id="results"
  />
</main>
`;
